pkg/forth/forth.go

```diff
@@ -157,39 +185,62 @@ func errRecover(errp *error) {
 	}
 }
 
-/* iEval takes a Forth and strings and splits the string on space
- * characters, pushing each element on the stack or invoking the
- * operator if it is found in the opmap.
- */
-func iEval(f Forth, s string) {
-	// TODO: create a separator list based on isspace and all the
-	// non alpha numberic characters in the opmap.
-	for _, val := range strings.Fields(s) {
-		//fmt.Printf("eval %s stack %v", val, f.Stack())
-		fun := opmap[val]
-		if fun != nil {
-			//fmt.Printf("Eval ...:")
-			fun(f)
-			//fmt.Printf("Stack now %v", f.Stack())
-		} else {
-			f.Push(val)
-			//fmt.Printf("push %s, stack %v", val, f.Stack())
+// Eval takes a Forth and []Cell, pushing each element on the stack or invoking the
+// operator if it is found in the opmap.
+func Eval(f Forth, cells ...Cell) (err error) {
+	defer errRecover(&err)
+	for _, c := range cells {
+		Debug("eval %s(%T) stack %v", c, c, f.Stack())
+		switch s := c.(type) {
+		case string:
+			fun := Getop(s)
+			if fun != nil {
+				Debug("Eval ... %v:", f.Stack())
+				fun(f)
+				Debug("Stack now %v", f.Stack())
+			} else {
```

Comment:
This is where you can check if the arg starts with a single quote which means it is a string. So:

- `swap` is the command `swap`
- `'swap` is the literal string `swap`
- `xswap` is the literal string `xswap` because there is no command called xswap

Reply:
done, see test